stdlib/private/StdlibCollectionUnittest/CheckRangeReplaceableCollectionType.swift

```diff
@@ -44,6 +51,78 @@ internal enum RangeSelection {
         return start..<end
     }
   }
+
+  internal func countableRange<C : Collection>(in c: C) -> CountableRange<C.Index> {
+    switch self {
+      case .emptyRange: return c.endIndex..<c.endIndex
+      case .leftEdge: return c.startIndex..<c.startIndex
+      case .rightEdge: return c.endIndex..<c.endIndex
+      case .middle:
+        let start = c.index(c.startIndex, offsetBy: c.count / 4)
+        let end = c.index(c.startIndex, offsetBy: 3 * c.count / 4 + 1)
+        return start..<end
+      case .leftHalf:
+        let start = c.startIndex
+        let end = c.index(start, offsetBy: c.count / 2)
+        return start..<end
+      case .rightHalf:
+        let start = c.index(c.startIndex, offsetBy: c.count / 2)
+        let end = c.endIndex
+        return start..<end
+      case .offsets(let offsets):
+        let start = c.index(c.startIndex, offsetBy: numericCast(offsets.lowerBound))
+        let end = c.index(c.startIndex, offsetBy: numericCast(offsets.upperBound))
+        return start..<end
+    }
+  }
+
+  internal func closedRange<C : Collection>(in c: C) -> ClosedRange<C.Index> {
+    switch self {
+      case .emptyRange: fatalError("Closed range cannot be empty")
+      case .leftEdge: return c.startIndex...c.startIndex
+      case .rightEdge: return c.endIndex...c.endIndex
+      case .middle:
+        let start = c.index(c.startIndex, offsetBy: c.count / 4)
+        let end = c.index(c.startIndex, offsetBy: 3 * c.count / 4 + 1)
+        return start...end
+      case .leftHalf:
+        let start = c.startIndex
+        let end = c.index(start, offsetBy: c.count / 2)
+        return start...end
+      case .rightHalf:
+        let start = c.index(c.startIndex, offsetBy: c.count / 2)
+        let end = c.endIndex
+        return start...end
+      case .offsets(let offsets):
+        let start = c.index(c.startIndex, offsetBy: numericCast(offsets.lowerBound))
+        let end = c.index(c.startIndex, offsetBy: numericCast(offsets.upperBound))
+        return start...end
+    }
```

Comment:
I'm confused by this as well. I put a `fatalError()` in my new test to see if it would trip, but while there was a warning during compilation, the test didn't actually seem to run. Is there more to adding a test than calling `self.test() { ... }`? I'm going to clean my build folder and build from scratch.